instagram/src/App.js

```diff
@@ -0,0 +1,81 @@
+import React, { Component } from 'react';
+import dummyData from './dummy-data';
+import PostContainer from './components/PostContainer/PostContainer';
+import SearchBar from './components/SearchBar/SearchBar';
+import './App.scss';
+
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      liked: false,
+      searchValue: '',
+      data: dummyData
+    };
+  }
+
+  liker = timestamp => {
+    this.setState({
+      liked: !this.state.liked
+    });
+
+    let filt = this.state.data.slice().filter(post => post.timestamp === timestamp).pop(),
+      negaFilt = this.state.data.slice().filter(post => post.timestamp !== timestamp);
+
+    filt.likes = this.state.liked === false ? filt.likes + 1 : filt.likes - 1;
+
+    negaFilt.unshift(filt);
+
+    this.setState({
+      data: negaFilt
+    })
+  }
```

Comment:
One thing that you need to be careful here is that you should avoid calling `setState` twice in one go like this. React handles `setState` in batches, so `setState` in line 18 and 29 will be run together.

Reply:
https://twitter.com/dan_abramov/status/887963264335872000?lang=en